src/eval.c

```diff
@@ -1854,32 +1864,39 @@ get_lval_subscript(
 	    glv_status = get_lval_dict_item(name, lp, key, len, &p, &var1,
 							flags, unlet, rettv);
 	    if (glv_status == GLV_FAIL)
-		return NULL;
+		goto done;
 	    if (glv_status == GLV_STOP)
 		break;
 	}
 	else if (v_type == VAR_BLOB)
 	{
 	    if (get_lval_blob(lp, &var1, &var2, empty1, quiet) == FAIL)
-		return NULL;
+		goto done;
 
 	    break;
 	}
 	else if (v_type == VAR_LIST)
 	{
 	    if (get_lval_list(lp, &var1, &var2, empty1, flags, quiet) == FAIL)
-		return NULL;
+		goto done;
 	}
 	else  // v_type == VAR_CLASS || v_type == VAR_OBJECT
 	{
 	    if (get_lval_class_or_obj(cl_exec, v_type, lp, key, p, flags,
 			quiet) == FAIL)
-		return NULL;
+		goto done;
 	}
+
+	clear_tv(&var1);
+	clear_tv(&var2);
     }
 
+    rc = OK;
+
+done:
     clear_tv(&var1);
-    return p;
+    clear_tv(&var2);
+    return rc == OK ? p : NULL;
```

Comment:
This got me wondering. Is it fine to call `clear_tv()` for `var1` and  `var2` twice? Once on line 1890+1891 and then here on line 1897 and 1898 again when the loop finished?

Reply:
Only simple types like number, string, bool and float can be used to index a List, Dict, Blob, Object or a Class.
Other composite types are not supported as index values.  With the simple types, clear_tv() can be called twice
without causing any reference counting issues.  When encountering an unsupported composite type, this 
function will jump to the end of the function, where the type is cleared only once.

Reply:
Patch 8.0.0352 (https://github.com/vim/vim/commit/f06e5a549f42396be3478ccc1b5f03be64e1173e) is related to this.

Reply:
okay thanks